msal/application.py

```diff
@@ -562,6 +588,7 @@ def initiate_device_flow(self, scopes=None, **kwargs):
             - an error response would contain some other readable key/value pairs.
         """
         return self.client.initiate_device_flow(
+            headers={'client-request-id': _get_new_correlation_id()},
```

Comment:
Don't you mean to do this instead?
```suggestion
            headers={
                'client-request-id': _get_new_correlation_id(),
                'x-client-current-telemetry': _build_current_telemetry_request_header(
                    self.INITIATE_DEVICE_FLOW),
                },
```

Reply:
telemetry data is only sent to the /authorize or /token endpoint, so in this case we wouldn't include the header.

Reply:
In that case, we might as well completely remove that `INITIATE_DEVICE_FLOW` api id definition.

Reply:
You're right - no need for INITIATE_DEVICE_FLOW. Also added comment